test/transformation-test.coffee

```diff
@@ -35,7 +35,11 @@ parseApiBlueprint = (source, type, cb) ->
 
     sourcemap = result?.sourcemap
 
-    err = adapter.transformError(source, err)
+    if err
+      # Protagonist does not include error in parse result, only in err
+      result.error = err
```

Comment:
`protagonist.parse` will throw for type errors as out is now (https://github.com/apiaryio/protagonist/blob/master/src/parse_async.cc#L37-L58). There is separate handling for user vs programming errors.

From Apiary side, the parse result from the API (serialised to/from JSON) will already contain the error as the format in API Blueprint AST (https://github.com/apiaryio/api-blueprint-ast/blob/master/Parse%20Result.md#error) as demonstrated in [API Blueprint AST test fixture](https://github.com/apiaryio/apiary/blob/b58276c7293e3059b46852fcc84a024b294889d8/test/fixtures/parser/invalid-api-blueprint.coffee#L27).